enarx-keep/tests/bin/clock_gettime.c

```diff
@@ -0,0 +1,21 @@
+#include <sys/syscall.h>
+#include <time.h>
+
+struct timespec t;
+
+void _start(void) {
```

Comment:
```suggestion
__attribute__((naked)) void _start(void) {
```

gives better generated code with `ud2` after the last syscall:

```asm
0000000000001000 <_start>:
    1000:	ba e4 00 00 00       	mov    $0xe4,%edx
    1005:	b9 01 00 00 00       	mov    $0x1,%ecx
    100a:	48 8d 05 ef 2f 00 00 	lea    0x2fef(%rip),%rax        # 4000 <t>
    1011:	48 89 c6             	mov    %rax,%rsi
    1014:	89 d0                	mov    %edx,%eax
    1016:	89 cf                	mov    %ecx,%edi
    1018:	0f 05                	syscall 
    101a:	b8 3c 00 00 00       	mov    $0x3c,%eax
    101f:	ba 00 00 00 00       	mov    $0x0,%edx
    1024:	89 d7                	mov    %edx,%edi
    1026:	0f 05                	syscall 
    1028:	90                   	nop
    1029:	0f 0b                	ud2    
```

Reply:
Thanks @haraldh , just updated this. What does the `ud2` get us? Why do we want an invalid opcode after the last syscall?

Reply:
@haraldh Looks like the naked function does not want to work with the strings:  `cannot allocate stack for variable s, naked function`.

Reply:
> @haraldh Looks like the naked function does not want to work with the strings: `cannot allocate stack for variable s, naked function`.

@lkatalin boooh.. bad compiler.. bad suggestion of me :-1:

Reply:
just add `ud2` manually

Reply:
@lkatalin `ud2` always causes a CPU fault. Basically, it is the manual way of guaranteeing that the `exit` syscall never returns without a CPU fault.

Reply:
Makes sense now!